### PR TITLE
Docker stuff

### PR DESCRIPTION
**Pull request by sophiakoulen**
1. There is now a docker-compose-prod.yml and a docker-compose-dev.yml
By default, the dev version is used.
To use the prod version: `export TRANSCENDANCE_MODE=prod`

2. You don't need to change much to continue working with dev.
The only change you need to do is **you need to delete any .env files that are in react-app/ and in nestjs/**
Only .env files located in env/ is relevant.
The kickstart.env and kickstart.env.old functionnality stays the same.

3. There is now a `make down` to stop the containers (this is useful if they're running in background)

4. Everything works now with HTTPS. HTTP requests are redirected to HTTPS.
You can configure HTTP_PORT and HTTPS_PORT in kickstart.env

### CODE REVIEW

File: Makefile
Comment (znichola) [inline Makefile:1]: please add make cert to the make file

Comment (znichola) [reply Makefile:1]: I mean it should be run automatically on a make

File: react-app/src/routes/UserLogin.tsx
Comment (znichola) [inline react-app/src/routes/UserLogin.tsx:1]: made thing about this issue

**Review by znichola — APPROVED** [review]
looks good, lots of lines of code, I approve.

**Review by rnbw8 — APPROVED** [review]
magnificent improvements, thanks a lot

**Comment by sophiakoulen** [comment]
I added the rule to make the certificates in 'up'

**Comment by znichola** [comment]
@Scelewyn could you resolve this conflict, then I think we should do the merge and go from there

**Review by BPhilagor — APPROVED** [review]
Super !

**Review by Scelewyn — APPROVED** [review]
Insane work !

**Pull request merged**